lib/utils/docker.ts

```diff
@@ -173,3 +171,120 @@ export async function isBalenaEngine(docker: dockerode): Promise<boolean> {
 		dockerVersion.Engine && dockerVersion.Engine.match(/balena|balaena/)
 	);
 }
+
+export interface ExtendedDockerOptions extends dockerode.DockerOptions {
+	docker?: string; // socket path, e.g. /var/run/docker.sock
+	dockerHost?: string; // host name or IP address
+	dockerPort?: number; // TCP port number, e.g. 2375
+}
+
+export async function getDocker(
+	options: ExtendedDockerOptions,
+): Promise<dockerode> {
```

Comment:
This block of code in `docker.ts` is a conversion of `docker-js.js` (deleted by this PR) from JavaScript to TypeScript.